cpp/mg_utility/mage.hpp

```diff
@@ -0,0 +1,2757 @@
+
+ private:
+  std::string message_;
+};
+
+class NotEnoughMemoryException : public std::exception {
+ public:
+  const char *what() const throw() { return "Not enough memory!"; }
+};
+
+namespace util {
+// uint to int conversion in C++ is a bit tricky. Take a look here
+// https://stackoverflow.com/questions/14623266/why-cant-i-reinterpret-cast-uint-to-int
+// for more details.
+template <typename TDest, typename TSrc>
+TDest MemcpyCast(TSrc src) {
+  TDest dest;
+  static_assert(sizeof(dest) == sizeof(src), "MemcpyCast expects source and destination to be of the same size");
+  static_assert(std::is_arithmetic<TSrc>::value, "MemcpyCast expects source to be an arithmetic type");
+  static_assert(std::is_arithmetic<TDest>::value, "MemcpyCast expects destination to be an arithmetic type");
+  std::memcpy(&dest, &src, sizeof(src));
+  return dest;
+}
+}  // namespace util
+
+// Forward declarations
+class Nodes;
+using GraphNodes = Nodes;
+class GraphRelationships;
+class Relationships;
+class Node;
+class Relationship;
+struct MapItem;
+class Duration;
+class Value;
+
+mgp_memory *memory;
+
+/* #region Graph (Id, Graph, Nodes, GraphRelationships, Relationships, Properties & Labels) */
+
+/// Wrapper for int64_t IDs to prevent dangerous implicit conversions.
+class Id {
+ public:
+  Id() = default;
+
+  /// Construct Id from uint64_t
+  static Id FromUint(uint64_t id) { return Id(util::MemcpyCast<int64_t>(id)); }
+  /// Construct Id from int64_t
+  static Id FromInt(int64_t id) { return Id(id); }
+
+  int64_t AsInt() const { return id_; }
+  uint64_t AsUint() const { return util::MemcpyCast<uint64_t>(id_); }
+
+  bool operator==(const Id &other) const { return id_ == other.id_; }
+  bool operator!=(const Id &other) const { return !(*this == other); }
+
+  bool operator<(const Id &other) const { return id_ < other.id_; }
+
+ private:
+  explicit Id(int64_t id) : id_(id) {}
+
+  int64_t id_;
+};
+
+/// @brief Wrapper class for @ref mgp_graph.
+class Graph {
+ private:
+  friend class Node;
+  friend class Relationship;
+
+ public:
+  explicit Graph(mgp_graph *graph) : graph_(graph) {}
+
+  /// @brief Returns the graph order (number of nodes).
+  int64_t order() const;
+  /// @brief Returns the graph size (number of relationships).
+  int64_t size() const;
+
+  /// @brief Returns an iterable structure of the graph’s nodes.
+  GraphNodes nodes() const;
+  /// @brief Returns an iterable structure of the graph’s relationships.
+  GraphRelationships relationships() const;
+
+  /// @brief Returns the graph node with the given ID.
+  Node GetNodeById(const Id node_id) const;
+
+  /// @brief Returns whether the graph contains a node with the given ID.
+  bool Contains(const Id node_id) const;
+  /// @brief Returns whether the graph contains the given node.
+  bool Contains(const Node &node) const;
+  /// @brief Returns whether the graph contains the given relationship.
+  bool Contains(const Relationship &relationship) const;
```

Comment:
I chose to rename the methods like you suggested, with the caveat that there’s no `ContainsRelationship(const Id relationship_id)` because of the corresponding function (`mgp_graph_get_edge_by_id`) not existing in the C API.